server/src/main/java/org/opensearch/index/IndexSettings.java

```diff
@@ -918,6 +918,9 @@ private void setSearchSegmentOrderReversed(boolean reversed) {
     }
 
     private void setSearchIdleAfter(TimeValue searchIdleAfter) {
+        if (this.replicationType == ReplicationType.SEGMENT && this.getNumberOfReplicas() > 0) {
+            logger.warn("Search idle is disabled for indices with replicas using the Segment Replication strategy");
```

Comment:
More nitpicks, but I'd match the terminology here to the settings API. Something like: "Search idle is not supported for indices with replicas using 'replication.type: SEGMENT'"

Reply:
updated.